### ISSUE TITLE
Bot doesn't respond

### ISSUE THREAD

**Issue by jzaefferer** [huafu/hubot-gitter2#9]
I've installed and started the bot using the README instructions, so far so good. When starting it logs a bunch of stuff about rooms it found/joined(?) as well as users. When trying to interact with the bot, for example typing "[nameofbot] help", it outputs something like this:

``` sh
[hubot-gitter2.debug][object GitterClient<[sha]>] found user with id `[other-sha]`: [object GitterUser<jzaefferer>]
```

Otherwise nothing happens. Googling around didn't yield any results, so I'm hoping for some guidance here.

**Comment by alexandru-calinoiu** [comment]
+1

**Comment by alexandru-calinoiu** [comment]
The issue that I had it was that I was using the gitter token from the same user I was trying to issue commands from. This makes sens since the bot should not talk to himself.

**Comment by jzaefferer** [comment]
That's an interesting hint, thank you. Will see if that was the same issue I had, its been a while.

**Comment by jzaefferer** [comment]
I signed up for another GitHub account, used that to sign up for another Gitter account, got the token from that, started the bot with the new token, and got the same result. The bot doesn't respond to anything, and I can't tell what I'm doing wrong.

@alexandru-calinoiu since you got it working, could you write down a few more details than what's in the readme? That would help a lot. It seems possible after all...

**Comment by alexandru-calinoiu** [comment]
The only thing that I noticed is that when I add him to a new room he does not reply, to fix that I have to restart it and will work.

Nothing else actually, could you do a log dump maybe we can see something.

**Comment by raine** [comment]
Same problem. I have a simple ping script and no get no reply.

Log says:

```
[hubot-gitter2.debug][object GitterClient<189b95ef8bec5187da92dde95b92af41a2e8ce0d>] found user with id `529c6acded5ab0b3bf04d461`: [object GitterUser<raine>]
```

**Comment by raine** [comment]
It looks like `bot.hear` works but when using `bot.respond` it doesn't register that messages are directed to it.

**Comment by raine** [comment]
Okay I realized now that I had the wrong name configured for the bot and expected it to obey its github username!

**Comment by jzaefferer** [comment]
Thanks @raine. By setting `--name ...` I finally got the bot working. Created #11 to address that in the readme.

**Comment by prtksxna** [comment]
I am facing this problem again, and setting the `--name` to the correct username doesn't seem to be helping.

**Comment by EdwinChua** [comment]
For anyone who's encountering this issue, try to talk to hubot with another Gitter account.

Steps:
- I registered with my Github account to get the API key for Gitter
- Then, I used my Twitter account to register another Gitter account
- With both accounts in the same room, my hubot responded via my Github-created account, only when messaged with my Twitter-created one.

It wasn't obvious to me that the bot would not respond to it's user name (i.e. the Github account)
